src/legacy/ui/public/test_harness/test_harness.js

```diff
@@ -29,6 +30,16 @@ import './test_harness.css';
 import 'ng_mock';
 import { setupTestSharding } from './test_sharding';
 
+const mockNewPlatformCommonPlugins = () => {
+  sinon.replace(npSetup, 'plugins', {
```

Comment:
@streamich  @lukeelmers  looks like you forgot to add npSetup mock for our legacy karma tests.

Reply:
@alexwizp The new platform changes LGTM, but I'm not 100% sure this is how we should address mocks in the karma tests. Since NP mocks are only being provided for jest, I'd prefer to convert tests to jest instead, and only keep karma ones for stuff that hasn't been deangularized yet and still needs ngMock. Or simply to add mocks to each of the tests that require them if there aren't too many.

@streamich what do you think?

cc @elastic/kibana-platform

Reply:
@lukeelmers @spalger @timroes @dmlemeshko   Not sure that it's possible in a short period. Let me explain. Please open the **test_harness.js** file and go to: 
![image](https://user-images.githubusercontent.com/20072247/62109345-c0cc9080-b2b4-11e9-9cba-e31141c58b1a.png)
As you see during bootstrap we execute all registered '**hacks**'.  It happens before executing the test cases. It's a tricky point because in case of exception in some of 'hack' function we cannot initialize/bootstrap correctly all needed dependencies.

In our case we faced with exception because  **vis_type_markdown** hack function has uninitialized reference to **npSetup.plugins.data* as a result we have a lot of unpredictable non-working tests.

What we can do now: 
1) Keep it as it or move mock to other place if you think that it's not a good place for it. I think for these type of tests is enough to mock only 'registerFunction' function
2) Instead of npSetup.plugins.data use approach with direct import from 'nterpreter/public/registries'. 
3) Try to add mocks to each of the tests. But not sure that tests should know about platform. e.g. see changes for markdown_fn.test.ts. I've removed extra import and think we shouldn't add it in future.

Reply:
Logic was moved to the separate file

Reply:
Alternative solution: https://github.com/elastic/kibana/pull/42261/files
Thanks @streamich